sql/core/src/main/scala/org/apache/spark/sql/execution/SparkStrategies.scala

```diff
@@ -72,6 +72,8 @@ abstract class SparkStrategies extends QueryPlanner[SparkPlan] {
         case Limit(IntegerLiteral(limit), Project(projectList, Sort(order, true, child)))
             if limit < conf.topKSortFallbackThreshold =>
           TakeOrderedAndProjectExec(limit, order, projectList, planLater(child)) :: Nil
+        case Limit(IntegerLiteral(limit), child) if plan.isStreaming =>
+          StreamingLimitExec(limit, planLater(child)) :: Nil
```

Comment:
I didn't need to create a separate Strategy for this because there's already this Strategy for limits, but I can create a different one if needed.